table/block_based_table_factory.cc

```diff
@@ -47,6 +47,12 @@ BlockBasedTableFactory::BlockBasedTableFactory(
   if (table_options_.index_block_restart_interval < 1) {
     table_options_.index_block_restart_interval = 1;
   }
+  if (table_options_.partition_filters &&
```

Comment:
We do still allow partitioning indexes without partitioning filters.

If it turns out that most users who partition indexes would partition filters too (which I am guessing it is the case) we should at some point merge them to one parameter.